File: src/Application.Client/src/types/message.ts
export interface Message {
  id: string;
  senderId: number;
  senderName: string;
  senderImageUrl: string;
  recipientId: number;
  recipientName: string;
  recipientImageUrl: string;
  content: string;
  dateRead?: string;
  messageSent: string;
  currentUserSender?: boolean;
}

File: src/Application.Domain/Entities/Message.cs
namespace Application.Domain.Entities;

public class Message
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public required string Content { get; set; }
    public DateTime? DateRead { get; set; }
    public bool SenderDeleted { get; set; }
    public bool RecipientDeleted { get; set; }
    public DateTime MessageSent { get; set; } = DateTime.UtcNow;

    public string SenderId { get; set; } = null!;
    public User Sender { get; set; } = null!;
    public string RecipientId { get; set; } = null!;
    public User Recipient { get; set; } = null!;
}
